core/src/main/java/org/apache/iceberg/DataFiles.java

```diff
@@ -134,6 +134,8 @@ public static class Builder {
     private Map<Integer, ByteBuffer> upperBounds = null;
     private ByteBuffer keyMetadata = null;
     private List<Long> splitOffsets = null;
+    private List<Integer> equalityFieldIds = null;
```

Comment:
add a setter for `equalityFieldIds` so that the parser unit test can cover this field too.